Scripts/Runtime/Data/Collections/DeferredNativeArray.cs

```diff
@@ -243,8 +254,6 @@ public unsafe JobHandle Clear(JobHandle inputDeps)
         /// <param name="nativeArrayOptions">The <see cref="NativeArrayOptions"/> for initializing the array memory.</param>
         public unsafe NativeArray<T> DeferredCreate(int newLength, NativeArrayOptions nativeArrayOptions = NativeArrayOptions.ClearMemory)
         {
-            AssertForDeferredCreate();
```

Comment:
Potential issues now that this assert is gone.
1. Calling DeferredCreate multiple times in a row will leak memory
2. There is no tracking whether a clear has been scheduled between the deferred create calls so you could be leaking memory even though you think you're not calling deferred created multiple times in a row
3. `ClearJob` holds a pointer to `m_BufferInfo` instead of making a copy of it so right now it will clear the most recently allocated memory not the currently allocated memory at the time of schedule. This is another opportunity to leak memory.

Reply:
1. Added an assert and flag to ensure you can't call this multiple times in a row now.
2. Via the flag the tracking is here so you can't leak because you can't allocate again until you've cleared.
3. This would be awesome but we can't actually do it because the copy of the buffer info wouldn't have the inner data exist until later. With the flag and assert we're protected against leaks.